### PR TITLE
Drag and drop to upload

### PR DESCRIPTION
**Pull request by jbidoret**
Add support for Drag and drop files within the media panel.
A progressive enhancement that triggers the regular file upload field, by filling it with dropped files.

### CODE REVIEW

File: src/media.js [hunk 1]
```diff
@@ -4,3 +4,34 @@ window.addEventListener('load', () => {
...
     activateCloseSubmenus();
 });
+
+/**
+ * Drag and drop files anywhere to fill regular file input
+ */
+const fileInput = document.getElementById('file');
+const fileMenu = fileInput.closest('details');
+
+'drag dragstart dragend dragover dragenter dragleave drop'.split(' ').forEach(function(listener){
```
Comment (n-peugnet) [inline src/media.js:14]: I think I would prefer a simpler `for...of` construct. More like:

```js
const dragListeners = ['drag', 'dragstart', 'dragend', 'dragover', 'dragenter', 'dragleave', 'drop']
for (const listener of dragListeners) {
    // register listener
}
```

File: src/media.js [hunk 2]
```diff
@@ -4,3 +4,34 @@ window.addEventListener('load', () => {
...
+    document.body.addEventListener(listener,function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+    },false);
+});
+
+'dragover dragenter'.split(' ').forEach(function(listener){
```
Comment (n-peugnet) [inline src/media.js:21]: Same comment here.

**Review by n-peugnet — COMMENTED** [review]
Just a small comment

File: src/media.js (hunk 1 above)
Comment (jbidoret) [reply src/media.js]: Sure. Sorry for my js style :)

**Comment by jbidoret** [comment]
OK. It seems this last commit could pass the tests.
I added multiple drops ability, and modified the dirty string split to a cleaner `for of` loop.
Could you test it and tell me? (no urge!)

**Comment by vincent-peugnet** [comment]
> OK. It seems this last commit could pass the tests. I added multiple drops ability, and modified the dirty string split to a cleaner `for of` loop. Could you test it and tell me? (no urge!)

I did some test on my side and it works fine ! Thanks Julien !

I did tests on:

- Firefox
- Chromium

**Comment by vincent-peugnet** [comment]
For me it's good to go ! If nobody have any objection I'm going to merge it soon 😃

**Pull request merged**